chore(qa): prevent false positive exceptions on ClusteringRule#after

npepinpe opened the pull request:
## Description

- waits until all brokers are shut down before deleting their data, otherwise exceptions are thrown (and we don't want to ignore exceptions)
- also prevents a small memory leak where on restarting brokers, brokers are removed from the clustering rule, but kept in the closeable list, resulting in them not being garbage collected until the process exits (happens if you run your tests in a loop).
- minor formatting

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

Zelldon commented on qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java:
```diff
@@ -208,16 +207,40 @@ protected void before() throws IOException {
+    // wait until the brokers are shutdown before deleting their data, otherwise exceptions will be
+    // thrown and cause developers to waste time figuring out that they aren't production bugs
+    try {
+      latch.await(10, TimeUnit.SECONDS);
```
Do you think 10 seconds is enough? Maybe we use the stepTimeout from the cfg?

Zelldon commented on qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java:
```diff
@@ -208,16 +207,40 @@ protected void before() throws IOException {
+
+    for (final var base : brokerBases.entrySet()) {
+      try {
+        FileUtil.deleteFolder(base.getValue().getAbsolutePath());
```
Who did this before?

Zelldon replied:
Ah I see it is done in the closeables. But why not anymore? Brokers should be closed now right?

npepinpe replied:
I'm actually not sure why we keep track of the folders here and also in the closeables to be honest. My issue with closeables is basically that keeping track of resources differently (i.e. one is the resource itself, another is a closure capturing it) led to out of memory errors for me (since we removed things from `brokerBase` and `brokers` but not closeables).

So like I said, I'm actually not sure what the gain of closeables is for the brokers if we keep track of those values in another map anyway :sweat_smile:

Zelldon replied:
Furthermore if we use a temp folder rule, this should cleanup by himself right?

Zelldon commented on qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java:
```diff
@@ -511,6 +532,7 @@ public void stopBroker(final int nodeId) {
           broker.getConfig().getNetwork().getCommandApi().getAddress();
       final List<Integer> brokersLeadingPartitions = getBrokersLeadingPartitions(socketAddress);
       broker.close();
+      closeables.remove(broker);
```
nice catch

Zelldon replied:
So close was also called twice? But this should be cleared after the test class?

Ah ok and if you run a test in a loop this piles up right

Zelldon replied:
Where do you see that the brokers are added to the closeables? Haven't found this in the code?

npepinpe replied:
Uh, you're right, they aren't. I added it because I ran into this issue in the past and had to do that, apparently it was already removed, my bad :sweat_smile: Makes more sense to not have the brokers managed by closeables anyway

Zelldon commented on qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java:
```diff
@@ -208,16 +207,40 @@ protected void before() throws IOException {
+              latch.countDown();
+            });
 
-    brokers.values().parallelStream().forEach(Broker::close);
```
Ah the parallel stream directly returns after beeing called?

npepinpe commented:
Alright, so my fix worked but for 100% the wrong reasons.

The reason this was happening is that we modify the test statement with `closeables.apply` in the `ClusteringRule#apply` method - so closeables were actually being closed before the ClusteringRule. So thank god for your review, because I also assumed `*.parellelStream().forEach()` was synchronous and was kind of confused. I'll revert it.

npepinpe commented:
I updated it with the correct fix - again sorry for the waste of time

npepinpe commented:
I guess we could also do that. I think we agree that having two ways is the problem :sweat_smile: 

I'm not sure which is better, tbh, and I don't have a strong opinion.

Zelldon commented:
me2 would be just nice to have one way :+1:

npepinpe commented:
I removed the `brokerBases` map and switched to using a fixed path with a `TemporaryFolder` rule. I guess we just have to remember to compose rules using the order of statement `apply`s.

npepinpe commented:
bors r+

npepinpe commented:
bors retry
